yang/IETF-04/ietf-vpn-common.yang

```diff
@@ -0,0 +1,505 @@
+  }
+
+/* Grouping */
+
+grouping vpn-profile-cfg {
+    container valid-provider-identifiers {
+      list cloud-identifier {
+        if-feature "cloud-access";
+        key "id";
+        leaf id {
+          type string;
+          description
+            "Identification of cloud service.
+             Local administration meaning.";
+        }
+        description
+          "List for Cloud Identifiers.";
+      }
+      list encryption-profile-identifier {
+        key "id";
+        leaf id {
+          type string;
+          description
+            "Identification of the SP encryption profile
+             to be used.  Local administration meaning.";
+        }
+        description
+          "List for encryption profile identifiers.";
+      }
+      list qos-profile-identifier {
+        key "id";
+        leaf id {
+          type string;
+          description
+            "Identification of the QoS Profile to be used.
+             Local administration meaning.";
+        }
+        description
+          "List for QoS Profile Identifiers.";
+      }
+      list bfd-profile-identifier {
+        key "id";
+        leaf id {
+          type string;
+          description
+            "Identification of the SP BFD Profile to be used.
+             Local administration meaning.";
+        }
+        description
+          "List for BFD Profile identifiers.";
+      }
+      list forwarding-profile-identifier {
+        key "id";
+        leaf id {
+          type string;
+          description
+            "Identification of the Forwrding Profile Filter to be used.
+             Local administration meaning.";
+        }
+        description
+          "List for Forwrding Profile identifiers.";
+      }
+      list routing-profile-identifier {
+        key "id";
+        leaf id {
+          type string;
+          description
+            "Identification of the routing Profile to be used
+             by the routing-protocols within sites, vpn-
+             network-accesses or vpn-nodes for refering
+			 vrf-import/export policies.
+             This identifier has a local meaning.";
+        }
+        description
+          "List for Routing Profile Identifiers.";
+      }
+      nacm:default-deny-write;
+      description
+        "Container for Valid Provider Identifies.";
+    }
+    description
+      "Grouping for VPN Profile configuration.";
+  }
+
+  grouping status-timestamp {
+    leaf status {
+      type identityref {
+        base operational-status;
+      }
+      description
+        "Operations status";
+    }
+    leaf timestamp {
+      type yang:date-and-time;
+      description
+        "Indicates the actual date and time when
+         the service actually started (UP) or
+         stopped (DOWN).";
+    }
+    description
+      "This grouping defines some operational
+       parameters for the service.";
+  }
+
+  grouping service-status {
+    container status {
+      leaf admin-status {
```

Comment:
I thought we agreed to have the timestamp for admin-status, but as an optional attribute.